apps/log/lib/log/model/log.ex

```diff
@@ -61,7 +61,7 @@ defmodule Helix.Log.Model.Log do
     %__MODULE__{}
     |> cast(params, @creation_fields)
     |> validate_required(@required_fields)
-    |> put_primary_key()
+    |> put_change(:log_id, PK.pk_for(__MODULE__))
```

Comment:
It's actually pretty simple what it does:

When you say `Repo.something(input)`, if `input` is a struct, it simply reads the schema data and executes `something` to `input`. If `input` is a changeset, it'll include on the changeset the `Repo` and what operation (ie: `something`) it is executing, then execute the `prepare_changes` callbacks that are stored on the changeset, then separate all the associations from the changeset, ensure that associations of the kind `belongs_to` exists in the database (or insert/update them otherwise), then execute `something` on `input`, then ensure that all associations of the kind `has_many` and `has_one` exists in the database (or insert/update/delete them otherwise)

Basically, it checks the schema of the going-to-be-inserted record and ensures that after the transaction it is exactly what you specified (by lazily binding every association on the database)